aten/src/ATen/native/cuda/MultiTensorApply.cuh

```diff
@@ -294,6 +341,113 @@ void multi_tensor_apply(
   }
 }
 
+template <int depth, typename T, typename... ArgTypes>
+void multi_tensor_apply(
+    std::vector<std::vector<at::Tensor>>& tensor_lists,
+    T callable,
+    ArgTypes... args) {
+  // Note: [static arg vs. dynamic arg]
+  // Due to the dynamic nature of the workload, the kernel arguments aren't
+  // guaranteed to fit in the static 4kb kernel argument memory. Previously
+  // with the apex implementation, we overcame this limitation by dividing a
+  // multi_tensor_apply workload into multiple kernel launches. However, this
+  // led to low sustained occupancy, affecting the performance of memory bound
+  // ops.
+  //
+  // Based on the observation that the kernel argument memory limitation
+  // doesn't correlate well with available SM resources, we have adopted a
+  // different approach. When the kernel arguments fit into the static kernel
+  // argument memory, we use this memory to transfer the arguments. Conversely,
+  // when the kernel arguments don't fit into the static kernel argument
+  // memory, instead of sacrificing sustained occupancy, we use a page-locked
+  // cudaMemcpyAsync to transfer the arguments, then perform the entire
+  // workload in a single kernel.
+  if (can_use_static_tensor_list_meta(tensor_lists, depth)) {
+    multi_tensor_apply_static<depth, T, ArgTypes...>(
+        tensor_lists, callable, args...);
+    return;
+  }
+
+  TORCH_CHECK(
+      tensor_lists.size() == depth,
+      "Number of tensor lists has to match the depth.");
+  const size_t n_tensors = tensor_lists[0].size();
+
+  TORCH_CHECK(
+      n_tensors < static_cast<size_t>(std::numeric_limits<int>::max()),
```

Comment:
Why "int" is the type checked here? Use a specific bitwidth to ensure it is consistent across os.

Reply:
Good catch. I updated the type of `block_to_tensor` to `int*` then `uint32_t*` and forgot to update here. I should just go with `size_t` here to eliminate the numeric_limit checks.